src/pubsub.c

```diff
@@ -615,6 +615,107 @@ void publishCommand(client *c) {
     addReplyLongLong(c,receivers);
 }
 
+typedef struct {
+    dict *subscription_dict;
+    long count;
+    sds pat;
+} pubsubSubscribersCmdArgs;
+
+static void pubsubReplySubscribersClientInfo(client *c, list *clients) {
+    listNode *ln;
+    listIter li;
+    listRewind(clients, &li);
+    while ((ln = listNext(&li)) != NULL) {
+        client *client = listNodeValue(ln);
+        addReplyMapLen(c, 3);
+        addReplyBulkCString(c, "name");
+        if (client->name) {
+            addReplyBulk(c, client->name);
+        } else {
+            addReplyBulkCBuffer(c, "", 0);
+        }
+        addReplyBulkCString(c, "id");
+        addReplyBulkLongLong(c, client->id);
+        addReplyBulkCString(c, "addr");
+        addReplyBulkCString(c, getClientPeerId(client));
+    }
+}
+
+static void pubsubReplySubscribers(client *c, pubsubSubscribersCmdArgs args) {
```

Comment:
a little bit odd to pass the argument by value and not reference.

Reply:
changed to pass by reference.